### PR TITLE
Content distribution via templates and slots

### LINKED ISSUE #48: Template with slot not getting rendered
### Version
@nuxt/content: 1.0.2
nuxt: 2.12.2

### Reproduction Link
https://codesandbox.io/s/nuxtcontent-demo-fp7i6?file=/content/index.md
### Steps to reproduce
When using the `template` tag together with named slots `<template slot="slotname">`, the content of the template does not get rendered.

### What is Expected?
The `<template slot="slotname">` should render properly.

### What is actually happening?
The `<template slot="slotname">` does not render properly. Whatever you put inside, it gets rendered as `Template Content`.

image

### PR DESCRIPTION
**Pull request by delapuente**
Fix #48

This PR modifies the JSON serializer to include the `content` AST from `template` tags and use it to create the [`scopedSlots` property](https://vuejs.org/v2/guide/render-function.html#createElement-Arguments) for the `createElement` function. Due to parser limitations, the content cannot use templating nor [slot props](https://vuejs.org/v2/guide/components-slots.html#Scoped-Slots).

This PR also updates the documentation and example to illustrate this feature.

I'm still working on the tests.

### CODE REVIEW

File: lib/templates/nuxt-content.js [hunk 1]
```diff
@@ -37,7 +67,42 @@ function processNode (node, h, doc) {
...
+}
+
+function isTemplate(node) {
+  return node.tag === 'template'
+}
+
+function getSlotName (node) {
+  let name = ''
+  for (const propName of Object.keys(node.props)) {
+    if (!propName.startsWith('v-slot:')) { return }
```
Comment (Atinux) [inline lib/templates/nuxt-content.js:101]: What about checking if it starts with ‘#’ as well to handle the shortcut?

File: lib/templates/nuxt-content.js [hunk 2]
```diff
@@ -29,6 +29,36 @@ function propsToData (props, doc) {
...
+    data.scopedSlots = data.scopedSlots || {}
+    const template = child
+    const name = getSlotName(child)
+
+    // The `'default'` template will be treated in the `processNode` path
+    if (name !== 'default') {
+      // Create a new slot from it
+      const vDomTree = template.content.map(tmplNode => processNode(tmplNode, h, doc))
+      data.scopedSlots[name] = function () { return vDomTree }
+
```
Comment (Atinux) [inline lib/templates/nuxt-content.js:55]: Extra line here

File: lib/templates/nuxt-content.js (hunk 1 above)
Comment (delapuente) [reply lib/templates/nuxt-content.js:101]: For some reason, I thought `rehype` did not parse it. It does. I'll change. Thanks for the heads up.

**Comment by Atinux** [comment]
Great work :)

Last think is test, I think they can be added on https://github.com/nuxt/content/blob/master/test/component.test.js

cc @benjamincanac

**Comment by delapuente** [comment]
This is done. Dealing with the generated HTML was a bit messy, I've opted for a regexp to deal with the whitespace. I hope it's enough.

**Comment by mathieunicolas** [comment]
Thank you very much !

**Comment by delapuente** [comment]
You're very welcome. I'm very excited about this project.

**Pull request merged**